pkg/controller/reconciler/sevicequota_reconciler.go

```diff
@@ -0,0 +1,74 @@
+package reconciler
+
+import (
+	"fmt"
+
+	"github.com/asdine/storm/v3"
+	"github.com/oslokommune/okctl/pkg/apis/okctl.io/v1alpha1"
+	"github.com/oslokommune/okctl/pkg/cfn"
+	clientCore "github.com/oslokommune/okctl/pkg/client/core"
+	"github.com/oslokommune/okctl/pkg/config/constant"
+	"github.com/oslokommune/okctl/pkg/controller/resourcetree"
+	"github.com/oslokommune/okctl/pkg/servicequota"
+	"github.com/pkg/errors"
+)
+
+// ServiceQuotaReconciler handles reconciliation for service quotas
+type ServiceQuotaReconciler struct {
+	commonMetadata *resourcetree.CommonMetadata
+	stateHandlers  *clientCore.StateHandlers
+
+	provider v1alpha1.CloudProvider
+}
+
+// SetStateHandlers sets the state handlers
+func (r *ServiceQuotaReconciler) SetStateHandlers(handlers *clientCore.StateHandlers) {
+	r.stateHandlers = handlers
+}
+
+// NodeType returns the relevant ResourceNodeType for this reconciler
+func (r *ServiceQuotaReconciler) NodeType() resourcetree.ResourceNodeType {
+	return resourcetree.ResourceNodeTypeServiceQuota
+}
+
+// SetCommonMetadata knows how to store common metadata on the reconciler. This should do nothing if common metadata is
+// not needed
+func (r *ServiceQuotaReconciler) SetCommonMetadata(meta *resourcetree.CommonMetadata) {
+	r.commonMetadata = meta
+}
+
+// Reconcile knows how to create, update and delete the relevant resource
+func (r *ServiceQuotaReconciler) Reconcile(node *resourcetree.ResourceNode) (result ReconcilationResult, err error) {
+	switch node.State {
```

Comment:
A state switch seems redundant here

Reply:
Just wanted to enumerate the scenarios the reconciler should consider, and I like switch statements more than if 🙈